### PR TITLE
DDF-2131: Reimplemented versioning (plus content versioning) in the c…

### PR DESCRIPTION
**Pull request by rzwiefel**
#### What does this PR do? 
Moves metacard versioning to the CatalogFramework and also implements Content Versioning.
This is a WIP.  
Still need to setup configadmin for turning off versioning. 
Still need itests. 

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?

@stustison 
@pklinef

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@kcwire
@coyotesqrl ( FIRSTTTTTT!)

### CODE REVIEW

File: catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java [hunk 1]
```diff
@@ -178,6 +179,22 @@ public UpdateStorageResponse update(UpdateStorageRequest updateRequest)
             }
         }
```
Comment (rzwiefel) [inline catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java]: This is fixing a bug for updated content items. Previously we did not update the resourceUri or resourceSize, this ensures they get set when updated.

File: catalog/core/catalog-core-standardframework/pom.xml [hunk 2]
```diff
@@ -221,13 +230,16 @@
                 <configuration>
                     <instructions>
                         <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+
                         <Embed-Dependency>
                             catalog-core-solr,
                             guava,
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-standardframework/pom.xml]: Guava shouldn't be embedded any longer.

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+            .expireAfterAccess(1, TimeUnit.HOURS)
+            .build();
+
+    public CreateRequest versionCreate(CreateRequest createRequest) {
+        if (!historyEnabled) {
+            return createRequest;
+        }
+        if (createRequest.getProperties()
+                .getOrDefault(ALREADY_VERSIONED, "")
+                .equals(true)) {
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java]: Your default value is an empty string whereas your expected value is a boolean, then you're doing an equals call to a boolean. Why not just...
```java
if (createRequest.getProperties().getOrDefault(ALREADY_VERSIONED, false)) {
...
```

File: catalog/core/catalog-core-standardframework/pom.xml (hunk 2 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-standardframework/pom.xml]: Yup. Was going to pull it out after I finish. Didn't want to muck with it while I was getting stuff to work.

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java [hunk 4]
```diff
@@ -1341,11 +1371,15 @@ public UpdateResponse update(UpdateStorageRequest streamUpdateRequest)
...
                     storage.commit(updateStorageRequest);
+                    if (historianTransactionKey != null) {
+                        historian.commit(historianTransactionKey);
+                    }
                 } catch (StorageException e) {
                     LOGGER.error("Unable to commit content changes for id: "
                             + updateStorageRequest.getId(), e);
                     try {
                         storage.rollback(updateStorageRequest);
+                        historian.rollback(historianTransactionKey);
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java]: Should wrap the call to `storage.rollback` in a try/catch so you can reliably call the `historian.rollback`.

**Comment by coyotesqrl** [comment]
LGTM

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java (hunk 3 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java]: done

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java (hunk 4 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/CatalogFrameworkImpl.java]: Moved the `historian.rollback` to a finally block. done

**Comment by rzwiefel** [comment]
Updated to fix qualified content items (previews, etc) and added unit tests.

**Comment by rzwiefel** [comment]
Reminder to self, Need to update catalog-plugin-versioning blueprint to remove the HistorianPlugin reference that got left in there.

**Pull request closed without merging**